.tools/readmes/scanner.py

```diff
@@ -110,39 +116,60 @@ def expand_entity(self, entity):
         return self.entities[entity]
 
     def snippet(self, example: Example, readme_folder, api_name: str):
+        if self.lang_name not in example.languages:
+            return None
+
         github = None
         tag = None
         tag_path = None
-        if self.lang_name in example.languages:
-            for ex_ver in example.languages[self.lang_name].versions:
-                if ex_ver.sdk_version == self.sdk_ver:
-                    github = ex_ver.github
-                    if github is not None:
-                        if ex_ver.excerpts:
-                            excerpt = ex_ver.excerpts[0]
-                            if excerpt.snippet_tags:
-                                tags = excerpt.snippet_tags
-                                for t in tags:
-                                    if api_name in t:
-                                        tag = t
-                                if tag is None:
-                                    tag = tags[0]
-                            elif excerpt.snippet_files:
-                                snippet_files = excerpt.snippet_files
-                                # TODO: Find the best (or all?) snippet files, not the first.
-                                full_path = snippet_files[0]
-                                tag_path = "/".join(full_path.split("/")[3:])
-                                if "cross-services" in full_path:
-                                    tag_path = "../cross-services/" + tag_path
-                        elif ex_ver.block_content:
-                            base_folder = f"{config.language[self.lang_name][self.sdk_ver]['base_folder']}/"
-                            if base_folder in github:
-                                tag_path = (
-                                    self._lang_level_double_dots() + github.split(base_folder, 1)[1]
-                                )
-                            else:
-                                tag_path = github
-        if tag and github is not None and tag_path is None:
+
+        ex_ver = first(
+            [
+                v
+                for v in example.languages[self.lang_name].versions
+                if v.sdk_version == self.sdk_ver
+            ]
+        )
+
+        if ex_ver is None:
+            return None
+
+        github = ex_ver.github
+        if github is None:
+            return None
+
+        if ex_ver.excerpts:
+            excerpt = first(ex_ver.excerpts)
+            if excerpt is not None:
+                if excerpt.snippet_tags:
+                    tags = excerpt.snippet_tags
+                    tags = [*filter(lambda t: api_name in t, tags)]
+
+                    # This form ensures that READMEs are always consistent,
+                    # but does not always give the "best" README link.
+                    tags = [*sorted(tags)]
```

Comment:
Sort before filter to get deterministic order.

Reply:
Done